kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ApiVersionsServiceImpl.java

```diff
@@ -89,15 +89,24 @@ private static void intersectApiVersions(String sessionId, ApiVersionsResponseDa
      * @param sessionId The sessionId.
      * @param key The key data from an upstream API_VERSIONS response.
      * @param apiKey The proxy's API key for this API.
-     * @param apiKeysShortFunction
+     * @param apiKeysShortFunction function to determine the maximum supported version for a particular API Key
      */
     private static void intersectApiVersion(String sessionId, ApiVersion key, ApiKeys apiKey, Function<ApiKeys, Short> apiKeysShortFunction) {
         short mutualMin = (short) Math.max(
                 key.minVersion(),
                 apiKey.messageType.lowestSupportedVersion());
         if (mutualMin != key.minVersion()) {
-            LOGGER.trace("{}: {} min version changed to {} (was: {})", sessionId, apiKey, mutualMin, key.maxVersion());
-            key.setMinVersion(mutualMin);
+            if (ApiKeys.PRODUCE.equals(apiKey)) {
+                // Kafka 4 removed support for ProduceRequest v0..v2 however there is an issue with libRDKafka versions <= XXX that meant this broke compression support
+                // https://issues.apache.org/jira/browse/KAFKA-18659 marks v0..v2 as supported versions however the broker will reject all uses of these old requests.
+                // The proxy needs to replicate this special case handling so we can proxy older libRDKafka based clients (just about anything that isn't Java)
```

Comment:
nit: as above, `v0..v2` as written could be misinterpreted upon first glance as a single version number rather than a range
```suggestion
                // Kafka 4 removed support for ProduceRequest v0-v2 however there is an issue with libRDKafka versions <= XXX that meant this broke compression support
                // https://issues.apache.org/jira/browse/KAFKA-18659 marks v0-v2 as supported versions however the broker will reject all uses of these old requests.
                // The proxy needs to replicate this special case handling so we can proxy older libRDKafka based clients (just about anything that isn't Java)
```
or
```suggestion
                // Kafka 4 removed support for ProduceRequest v0 to v2 however there is an issue with libRDKafka versions <= XXX that meant this broke compression support
                // https://issues.apache.org/jira/browse/KAFKA-18659 marks v0 to v2 as supported versions however the broker will reject all uses of these old requests.
                // The proxy needs to replicate this special case handling so we can proxy older libRDKafka based clients (just about anything that isn't Java)
```